### PR TITLE
Polars feature updates

### PR DESCRIPTION
**Pull request by abajpai15**
Added polars to datetime_column_profile and unstructured_labeler

### CODE REVIEW

File: dataprofiler/profilers/datetime_column_profile.py [hunk 1]
```diff
@@ -256,8 +257,7 @@ def _get_datetime_profile(cls, df_series: pd.Series) -> dict:
         profile: dict = dict()
         activated_date_formats: list = list()
         len_df = len(df_series)
-
-        is_row_datetime = pd.Series(np.full((len(df_series)), False))
+        is_row_datetime = pd.Series(np.full((len_df), False))
```
Comment (taylorfturner) [inline dataprofiler/profilers/datetime_column_profile.py:260]: nice catch

File: dataprofiler/profilers/unstructured_text_profile.py [hunk 2]
```diff
@@ -708,37 +709,54 @@ def _update_words(
...
+                        w.strip(string.punctuation)
+                        for w in row.str.to_lowercase().str.split(by=" ")
+                    ]
+                    for row in data
+                )
+            else:
+                words = (
+                    [w.strip(string.punctuation) for w in row.lower().split()]
+                    for row in data
+                )
```
Comment (taylorfturner) [inline dataprofiler/profilers/unstructured_text_profile.py:724]: pre-commit made it like this?

File: dataprofiler/profilers/unstructured_text_profile.py [hunk 3]
```diff
@@ -708,37 +709,54 @@ def _update_words(
...
-            words = ([w.strip(string.punctuation) for w in row.split()] for row in data)
+            if type(data) is pl.DataFrame:
+                words = (
+                    [w.strip(string.punctuation) for w in row.str.split(by=" ")]
+                    for row in data
+                )
+            else:
+                words = (
+                    [w.strip(string.punctuation) for w in row.split()] for row in data
+                )
```
Comment (taylorfturner) [inline dataprofiler/profilers/unstructured_text_profile.py:734]: 90% of this is similar ... could make this more modular. follow-up PR maybe or if you can crank it out quick can do this

File: dataprofiler/profilers/unstructured_text_profile.py [hunk 4]
```diff
@@ -748,14 +766,19 @@ def update(self, data: Series) -> TextProfiler:
 
         profile = dict(sample_size=len_data)
 
+        if type(data) is pl.Series:
```
Comment (taylorfturner) [inline dataprofiler/profilers/unstructured_text_profile.py:769]: make this an `isinstance(data, pl.Series)` instead

Comment (taylorfturner) [reply dataprofiler/profilers/unstructured_text_profile.py:769]: haven't tested but something like this with `isinstance`

**Review by micdavis — DISMISSED** [review]

File: dataprofiler/profilers/unstructured_text_profile.py (hunk 2 above)
Comment (abajpai15) [reply dataprofiler/profilers/unstructured_text_profile.py:724]: Yep

File: dataprofiler/profilers/unstructured_text_profile.py (hunk 4 above)
Comment (taylorfturner) [inline dataprofiler/profilers/unstructured_text_profile.py:769]: let's update this and the `_update_words` prior to merge

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by taylorfturner — APPROVED** [review]

**Review by ksneab7 — APPROVED** [review]

**Pull request merged**